### PR TITLE
use conda env file

### LINKED ISSUE #111: Use mirgecom's conda env file for conda packages
instead of the hardcoded list in install-conda-dependencies

### PR DESCRIPTION
**Pull request by matthiasdiener**
Closes #111.

### CODE REVIEW

File: install-conda-dependencies.sh [hunk 1]
```diff
@@ -1,52 +0,0 @@
...
-        conda install --yes pocl-cuda libhwloc=1
-    fi
-
-    conda install --yes pyvisfile pyopencl islpy clinfo pymetis cantera
-
-    # We need an MPI installation to build mpi4py.
-    # Install OpenMPI if none is available.
-    if ! command -v mpicc &> /dev/null ;then
-        conda install --yes openmpi openmpi-mpicc
-    fi
```
Comment (inducer) [inline install-conda-dependencies.sh:51]: Looks like this functionality is being lost?

File: install-conda-dependencies.sh [hunk 2]
```diff
@@ -1,52 +0,0 @@
...
-        conda install --yes "$package"
-    done
-
-else
-
-    if [[ $(uname) == "Darwin" ]]; then
-        conda install --yes pocl libhwloc=1
-    else
-        conda install --yes pocl-cuda libhwloc=1
-    fi
```
Comment (inducer) [inline install-conda-dependencies.sh:43]: Is anything now taking care of installing the right hwloc? (see also #73)

Comment (matthiasdiener) [reply install-conda-dependencies.sh:43]: No, but haven't we fixed the issue? See mirgecom/169.

File: install-conda-dependencies.sh (hunk 1 above)
Comment (matthiasdiener) [reply install-conda-dependencies.sh:51]: Yeah, but from my experience this creates more issues than it solves. Maybe just print a warning?

Comment (MTCam) [reply install-conda-dependencies.sh:51]: Out of curiosity, what MPI gets used in CI when the examples run?
`        mpiexec -n 2 python -m mpi4py ${example}`

**Comment by MTCam** [comment]
I love what you've done with this from a pseudo-user perspective. It is very clear what to do in your branch if you need to add packages.

File: install.sh [hunk 3]
```diff
@@ -17,7 +17,6 @@ usage()
   echo "  --env-name=NAME       Name of the conda environment to install to. (default=ceesd)"
   echo "  --modules             Create modules.zip and add to Python path."
   echo "  --branch=NAME         Install specific branch of mirgecom (default=master)."
-  echo "  --conda-pkgs=FILE     Install these additional packages with conda."
```
Comment (matthiasdiener) [inline install.sh:20]: Restore this option

Comment (matthiasdiener) [reply install.sh]: Restored.

File: install-conda-dependencies.sh (hunk 1 above)
Comment (matthiasdiener) [reply install-conda-dependencies.sh:51]: > Out of curiosity, what MPI gets used in CI when the examples run?
> ` mpiexec -n 2 python -m mpi4py ${example}`

We usually install them on the system:

```
[[ $(uname) == Linux ]] && sudo apt-get install -y openmpi-bin libopenmpi-dev
[[ $(uname) == Darwin ]] && brew install openmpi
```

Comment (matthiasdiener) [reply install-conda-dependencies.sh:51]: > Looks like this functionality is being lost?

Restored the check.

**Review by MTCam — APPROVED** [review]
lgtm yo.

**Comment by matthiasdiener** [comment]
good to go @inducer?

File: install-conda-dependencies.sh (hunk 2 above)
Comment (inducer) [reply install-conda-dependencies.sh:43]: Fair.

File: install-conda-dependencies.sh (hunk 1 above)
Comment (inducer) [reply install-conda-dependencies.sh:51]: LGTM.

**Review by inducer — APPROVED** [review]

**Pull request merged**